### DISCUSSION TITLE
HTTP -> HTTPS redirects not working

### DISCUSSION THREAD

**Discussion by SpaceNerden** [concourse/concourse#8691] [Help & Support]
## Summary

HTTPS redirects are not working.

## Steps to reproduce

- Install Concourse on docker
- Add a target with an HTTP URL/IP
- Works fine
- Add HTTPS redirect on reverse proxy
- Set pipeline
- Get confused and maybe panic
```
error: Unexpected Response
Status: 404 Not Found
Body: 
```
I also saw this when pausing/unpausing
```
error: Unexpected Response
Status: 405 Method Not Allowed
Body: Method Not Allowed
```
- Change HTTP URL/IP with HTTPS (Or disable Force SSL/HTTPS redirect)
- Everything works again

## Expected results
Redirect to HTTPS, maybe a prompt asking if the user wants to change the target API URL to HTTPS

## Actual results
An error with a slightly unhelpful response.

## Additional context
I apologize if I get some terminologies wrong. I'm very new to this space and am still learning. Please correct me if I mistook any of the words.

## Triaging info

* Concourse version: 7.9.1
* Did this used to work? Not sure, this is my first time using Concourse.

**Comment by 4x0v7** [comment]
Have you seen the docs here? https://concourse-ci.org/concourse-web.html#web-ingress
If you are using docker (without compose), you can set the `CONCOURSE_EXTERNAL_URL` environment variable to your HTTPS url.
With docker it would be `-e CONCOURSE_EXTERNAL_URL=https://yourdomain.com` as part of the command.
If you use docker compose you can set it in the yaml like
```yaml
services:
  concourse:
    ...
    environment:
      CONCOURSE_EXTERNAL_URL: https://yourdomain.com
```

**Comment by taylorsilva** [comment]
Converted this to an issue since it's likely not a bug
